tests/test_voice.py

```diff
@@ -18,182 +18,153 @@
+        self.assertEqual(self.voice.mime_type, 'audio/ogg')
+        self.assertEqual(self.voice.file_size, 9199)
 
     @flaky(3, 1)
     @timeout(10)
     def test_send_voice_all_args(self):
         message = self._bot.sendVoice(
             self._chat_id,
             self.voice_file,
-            duration=self.duration,
+            duration=self.voice.duration,
             caption=self.caption,
-            mime_type=self.mime_type,
-            file_size=self.file_size)
+            mime_type=self.voice.mime_type,
+            file_size=self.voice.file_size,
+            disable_notification=False)
 
         self.assertEqual(message.caption, self.caption)
 
         voice = message.voice
 
-        self.assertTrue(isinstance(voice.file_id, str))
+        self.assertIsInstance(voice.file_id, str)
         self.assertNotEqual(voice.file_id, '')
-        self.assertEqual(voice.duration, self.duration)
-        self.assertEqual(voice.mime_type, self.mime_type)
-        self.assertEqual(voice.file_size, self.file_size)
+        self.assertEqual(voice.duration, self.voice.duration)
+        self.assertEqual(voice.mime_type, self.voice.mime_type)
+        self.assertEqual(voice.file_size, self.voice.file_size)
 
     @flaky(3, 1)
     @timeout(10)
-    def test_send_voice_ogg_file(self):
-        message = self._bot.sendVoice(
-            chat_id=self._chat_id,
-            voice=self.voice_file,
-            duration=self.duration,
-            caption=self.caption)
+    def test_get_and_download_voice(self):
+        new_file = self._bot.getFile(self.voice.file_id)
 
-        self.assertEqual(message.caption, self.caption)
+        self.assertEqual(new_file.file_size, self.voice.file_size)
+        self.assertEqual(new_file.file_id, self.voice.file_id)
+        self.assertTrue(new_file.file_path.startswith('https://'))
 
-        voice = message.voice
+        new_file.download('telegram.ogg')
 
-        self.assertTrue(isinstance(voice.file_id, str))
-        self.assertNotEqual(voice.file_id, '')
-        self.assertEqual(voice.duration, self.duration)
-        self.assertEqual(voice.mime_type, self.mime_type)
-        self.assertEqual(voice.file_size, self.file_size)
-
-    @flaky(3, 1)
-    @timeout(10)
-    def test_send_voice_ogg_file_with_custom_filename(self):
-        message = self._bot.sendVoice(
-            chat_id=self._chat_id,
-            voice=self.voice_file,
-            duration=self.duration,
-            caption=self.caption,
-            filename='telegram_custom.ogg')
-
-        self.assertEqual(message.caption, self.caption)
-
-        voice = message.voice
-
-        self.assertTrue(isinstance(voice.file_id, str))
-        self.assertNotEqual(voice.file_id, '')
-        self.assertEqual(voice.duration, self.duration)
-        self.assertEqual(voice.mime_type, self.mime_type)
-        self.assertEqual(voice.file_size, self.file_size)
+        self.assertTrue(os.path.isfile('telegram.ogg'))
```

Comment:
(applies to all downloads)
Should we check against the local file, that they're the same?